src/libNetwork/P2PComm.cpp

```diff
@@ -376,133 +525,97 @@ void P2PComm::HandleAcceptedConnection(
             }
             read_length += n;
         }
+    }
 
-        // Check if this message has been received before
-        bool found = false;
-        {
-            lock_guard<mutex> guard(
-                P2PComm::GetInstance().m_broadcastHashesMutex);
-            vector<unsigned char> msg_hash(hash_buf, hash_buf + HASH_LEN);
-            found = (P2PComm::GetInstance().m_broadcastHashes.find(msg_hash)
-                     != P2PComm::GetInstance().m_broadcastHashes.end());
-            // While we have the lock, we should quickly add the hash
```

Comment:
This isn't that "quickly", I think. We hold the global lock - `m_broadcastHashesMutex` while we do blocking calls to read from the socket (`read()`).